docs/proposals/timeouts.md

```diff
@@ -0,0 +1,119 @@
+      ProviderConfiguration...ReadinessCheckConfiguration.Timeout=15min for every DeployItem. Similar problems might occur
+      for exports.
+    - Even if the overall timeout for a DeployItem is set, no specific timeout is set for the helm chart deployment with 
+      helm 3. What does this mean? How long might the deployer hang in this step and thereby preventing another 
+      reconciliation even after the timeout controller has set the state on failure? Or will the helm 3 deployment fail,
+      due to some other default settings. There are similar questions for the manifest and helm-manifest deployer, e.g.
+      what is the default timeout for the http client etc.?
+  - There are some really strange or even inconsistent implementations e.g. 
+    - Timeout in customreadinesscheck. Checks are executed one after the other in an undefined order. So it depends on 
+      that order how much time was already spend before which might result in a failure of a readiness check timeout.
+    - deleteTimeout is used in the context of manifest and helm manifest deployer when deleting orphaned resources during
+      an upgrade. This setting has nothing to do with the deletion of a DeployItem which is very confusing.
+
+- Pickup timeout: If there is a high load on the system, the pickup of particular DeployItems by a deployer is deferred 
+  because the worker threads might be exhausted and therefore their state is set on failure by the timeout controller. 
+  Just waiting some additional time would have been resulted in their proper processing. This resulted in problems for
+  the Steampunk team.
+
+
+## Solution proposal
+
+- Remove pickup timeout, set it on a very large value or add some DeployItem specific setting:
+  - The main reason for the pickup timeout was to resolve the problem, that if someone wrongly configures a DeployItem
+    such that no deployer would be responsible, that this DeployItem is finished. This use case is mainly relevant during
+    development. Comparing this with the problem of failed DeployItems in productive systems under heavy load, seems to
+    be a bad deal. Furthermore, for the development scenario, it is already possible to find out that a DeployItem
+    is not picked up, by its empty status and such DeployItems could be stopped by the interrupt annotation. 
+    Therefore, the removal of this timeout or setting it on a high value might be a good idea. 
+  - If we want to be more defencive, we could add a pickup timeout to the DeployItem specification itself. Then the 
+    customer could decide by itself if he wants this feature e.g. in his development scenario.
+
+- Reduce number of processing timeout settings
+  - Motivation: It is less complicated for a user to only set one timeout for the deployment or deletion of a DeployItem.
+    Even finding here the right value is already complicated but to take all the other current settings into account
+    is nearly impossible.
```

Comment:
I completely agree with that. Even if we have fine-granular timeout settings (which likely nobody will use), there should be an easy way to increase all of them at once for slow components.